src/VTUFiles.f90

```diff
@@ -393,13 +409,38 @@ SUBROUTINE hasData_VTUXMLFileType(fileobj,varName,varFormat,bool)
       INTEGER(SIK) :: i
       !
       bool=.FALSE.
-      DO i=1,fileobj%varNameListLength
-        IF (varName==fileobj%varNameList(i).AND. &
-          varFormat==fileobj%dataFormatList(i)) bool=.TRUE.
-      ENDDO
+      IF(ALLOCATED(fileobj%varNameList)) THEN
+        DO i=1,fileobj%numDataSet
+          IF (varName==fileobj%varNameList(i).AND. &
+            varFormat==fileobj%dataFormatList(i)) bool=.TRUE.
+        ENDDO
+      ENDIF
     ENDSUBROUTINE hasData_VTUXMLFileType
 !
 !-------------------------------------------------------------------------------
+!> @brief Checks if fileobj has a given file.
+!> @param fileobj input file object.
+!> @param fname Name of the file to check.
+!> @param bool Whether or not the data is present.
+!>
+!> Checks if the VTU File has the specified file (which is written to pvtu).
+!>
+    SUBROUTINE hasFile_VTUXMLFileType(fileobj,fname,bool)
+      CHARACTER(LEN=*),PARAMETER :: myName='hasFile_VTUXMLFileType'
+      CLASS(VTUXMLFileType),INTENT(INOUT) :: fileobj
+      CHARACTER(LEN=*),INTENT(IN) :: fname
+      LOGICAL(SBK),INTENT(INOUT) :: bool
+      INTEGER(SIK) :: i
+      !
+      bool=.FALSE.
+      IF(ALLOCATED(fileobj%fileList)) THEN
+        DO i=1,fileobj%numFiles
+          IF (fname==fileobj%fileList(i)) bool=.TRUE.
```

Comment:
This loop will continue to execute after bool has been set to true.

Modify to:
```
IF(fname == fileobj%fileList(i)) THEN
  bool=.TRUE.
  EXIT
ENDIF
```